client/state/themes/selectors.js

```diff
@@ -574,6 +574,18 @@ export function isThemePremium( state, themeId ) {
 }
 
 /**
+ * Whether a WPCOM theme given by its ID belongs to the Premium Squared bundle.
+ *
+ * @param  {Object} state   Global state tree
+ * @param  {Object} themeId Theme ID
+ * @return {Boolean}        True if the theme is in the Premium Squared bundle
+ */
+export function isPremiumSquaredTheme( state, themeId ) {
+	const theme = getTheme( state, 'wpcom', themeId );
+	return theme && includes( PREMIUM_SQUARED_THEMES, theme.stylesheet );
```

Comment:
> Have you considered using only themeId here for checking (i.e. without the premium/ prefix)?

I don't think this would be a good idea, since a theme ID may conflict with a free theme. They may not conflict at this very moment (I haven't checked in depth though), but they might in the future, so this is not a solid and future-proof implementation.

> Alternatively, as we discussed before, instead of a hard-wired list of theme slugs (or IDs), you could also consider checking if theme.author was either Automattic or Woo (that would mean you'd have to keep getTheme and state tho).

As discussed in Slack, I think it makes sense to keep them both (Calypso & wpcom) lists in sync, adding comments to cross link the resources and update both when needed. These lists may rarely change in the future, so I think this is okay.